gnucash/report/trep-engine.scm

```diff
@@ -1676,7 +1827,19 @@ be excluded from periodic reporting.")
 
     (define (add-split-row split cell-calculators row-style transaction-row?)
       (let* ((account (xaccSplitGetAccount split))
-             (reversible-account? (acc-reverse? account)))
+             (reversible-account? (acc-reverse? account))
+             ;; Returns the number of required arguments of a function
+             (fn-arity (lambda (fn) (car (procedure-minimum-arity fn))))
+             ;; Calls the calculator-function for the provided cell with either
+             ;; two arguments (calculator-function split transaction-row?) or only
+             ;; one argument  (calculator-function split) for legacy support.
+             (calc-fn (lambda (cell split transaction-row?)
+                              (if (= (fn-arity (vector-ref cell 1)) 1)
+                                  ((vector-ref cell 1) split)
+                                  ((vector-ref cell 1) split transaction-row?))))
```

Comment:
I wonder if it may be simplified to:
```scheme
  (calc-fn (lambda (cell . rest) (apply (vector-ref cell 1) rest))
```

Reply:
Thanks for the feedback! I will check that. I love it when you can turn several lines of my code into a single line. Scheme is pretty fascinating.

Reply:
Ha here is that match again! I like it but one issue I see is that it creates some unusual structure in the calculated-function definitions. I mean the double (lambda (s) (lambda (tr?)) instead of the more traditional (lambda (s tr?)). Since the calculated-functions are also part of custom reports, etc., it puts the onus on whoever is creating these custom reports to maintain this strange structure. 

Considering this complaint in the roadmap:
> Scheme is impenetrable to most programmers. Expecting users to be able to write reports in Scheme is completely unreasonable.

I don't know whether it's a good idea to make it look even more funky especially the custom definition part. Also in the long term, if eventually we deprecate support for one argument, then we are left with a bunch of existing custom-functions with this strange double lambda structure which we don't even need anymore.

My version (although not as clever) has the benefit of putting the code to support the 2 types of calculator-functions entirely in add-split-row, hidden from whoever is designing the custom reports. The calculator-functions themselves follow a more standard pattern. The only inconvenience of my version I can see is that it introduces the guile-specific function. I really can't judge how much of a problem it is using that function. I don't really know enough to make that sort of call of 'pure' scheme vs guile. I suppose it comes down to whether there is ever a chance we keep scheme without guile? I mean we are already using guile modules, right? For instance the match function, that already comes from a guile-specific module ice-9 , no?

So we probably need to see if others want to pitch in here... There is also my alternative implementation where we simply keep one argument and disable the running totals in the multiline context. I think it's a bit of a missed opportunity but I would be OK with it as well, not a big deal.

On a completely side note RE: the complaint in the roadmap, yes scheme is a bit unusual and possibly overkill for this, but to be honnest if we make the effort of putting more comments in these scm files including some explanation of scheme clever but unusual structures when we use them, I think it might help maintanability and invite more people to participate.

Reply:
My patch was simply a pure scheme-based idea.

Please hold off on it now and we'll think of a neat way to upgrade the trep-engine while maintaining backward compatibility, without needing to rely too much on guile's specific functions. Maybe the calculated-cells can be changed from a list to record with a `version` field and a `columns` list.

FWIW (ice-9 match) is macro-based and widely adopted by guile maintainers, and is pretty much standardised as SRFI-204 across many lisp implementations.

Reply:
This was resolved through PR #1618